canvasobject.go

```diff
@@ -105,3 +109,85 @@ type Tabbable interface {
 type Tappable interface {
 	Tapped(*PointEvent)
 }
+
+type DragItem interface {
+	MimeType() string
+	URI() URI
+}
+
+type DragItemCursor interface {
+	// Image returns the image for the given cursor, or nil if none should be shown.
+	// It also returns the x and y pixels that should act as the hot-spot (measured from top left corner).
+	Image() (image.Image, int, int)
+}
+
+// DragSource is implemented by CanvasObjects to indicate that they can be dragged and dropped
+// onto another CanvasObject which implements DropTarget.
+type DragSource interface {
+	// DragSourceBegin is invoked when a drag and drop is initiated on this DragSource.
+	// It should return the DragItems it is sourcing, and optionally a non-nil DragItemCursor
```

Comment:
Let's not bake cursors into the main abstraction. Perhaps the DragItem could optionally implement DragItemCursor? Or even just Cursorable?